src/sqlstorage.cc

```diff
@@ -43,45 +38,18 @@ void SQLStorage::storePrimaryPublic(const std::string& public_key) {
   request = kSqlGetSimple;
   req_response.clear();
   if (db.exec("SELECT count(*) FROM primary_keys;", callback, this) != SQLITE_OK) {
-    LOG_ERROR << "Can't get count of public key table: " << db.errmsg();
+    LOG_ERROR << "Can't get count of keys table: " << db.errmsg();
     return;
   }
   const char* req;
   if (boost::lexical_cast<int>(req_response["result"])) {
-    req = "UPDATE OR REPLACE primary_keys SET public = ?;";
+    req = "UPDATE OR REPLACE primary_keys SET (public, private) = (?,?);";
   } else {
-    req = "INSERT INTO primary_keys(public) VALUES (?);";
+    req = "INSERT INTO primary_keys(public,private) VALUES (?,?);";
   }
-  auto statement = db.prepareStatement<std::string>(req, public_key);
+  auto statement = db.prepareStatement<std::string>(req, public_key, private_key);
```

Comment:
Can't we use INSERT OR REPLACE INTO here?

Reply:
Ah, I see. Having a dummy column in the schema feels a bit weird (not say it is wrong, just weird), so the current solution is probably fine, for now at least

Reply:
Yes but encoding that the table has only one row in the schema could be nice for added safety/self documentation.

The second example here https://stackoverflow.com/a/33104119 looks nice.